src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.js

```diff
@@ -364,6 +374,24 @@ function ComposerWithSuggestions({
         [isKeyboardShown, isSmallScreenWidth, parentReportActions, report, reportActions, reportID, handleSendMessage, suggestionsRef, valueRef],
     );
 
+    const onChangeText = useCallback(
+        (commentValue) => {
+            updateComment(commentValue, true);
+
+            if (isIOSNative && syncSelectionWithOnChangeTextRef.current) {
+                const positionSnapshot = syncSelectionWithOnChangeTextRef.current.position;
+                syncSelectionWithOnChangeTextRef.current = null;
+
+                InteractionManager.runAfterInteractions(() => {
```

Comment:
Do you mean in code? It is required to ensure that the selection is set imperatively after all state changes are effective; without it, the problem would still persist since `setSelection` would run too eager.

Reply:
Sure thing. Resolved in 98436b74a0db72abc683e0f35799acbbba9874ab